tests/Unit/EventListener/TagListenerTest.php

```diff
@@ -48,7 +56,22 @@ public function setUp()
             SymfonyResponseTagger::class
         );
 
-        $this->listener = new TagListener($this->cacheManager, $this->symfonyResponseTagger);
+        $this->cacheableRule = new RuleMatcher(
+            new CacheableRequestMatcher(),
+            new CacheableResponseMatcher()
+        );
+
+        $this->mustInvalidateRule = new RuleMatcher(
+            new UnsafeRequestMatcher(),
+            new NonErrorResponseMatcher()
```

Comment:
strictly speaking, a unit test should mock the rule matcher for the behaviour we expect and then have separate unit tests for each matcher...